pymmcore_plus/core/_mmcore_plus.py

```diff
@@ -724,6 +725,115 @@ def setShutterOpen(self, *args):
             state = args
         self.events.propertyChanged.emit(shutterLabel, "State", state)
 
+    def deleteConfig(self, group: str, preset: str) -> None:
+        super().deleteConfig(group, preset)
+        self.events.presetDeleted.emit(group, preset)
+
+    def deleteConfigGroup(self, group: str) -> None:
+        super().deleteConfigGroup(group)
+        self.events.groupDeleted.emit(group)
+
+    def deletePresetDeviceProperties(
+        self, group: str, preset: str, device_property_list: List[Tuple[str, str]]
+    ) -> None:
+        """
+        Delete the specified list of (device, property) from the specified preset.
+
+        This method will emit the "newGroupPreset" signal with group and preset info.
+
+        Parameters
+        ----------
+        group: str
+            the group name.
+        preset: str
+            the preset name.
+        device_property_list: List[Tuple[str, str]]
+            list of (device, property) that will be removed from the preset.
+        """
+        dev_prop_val = [(k[0], k[1], k[2]) for k in self.getConfigData(group, preset)]
+
+        super().deleteConfig(group, preset)
+
+        _to_emit = []
+        for d, p, v in dev_prop_val:
+            if (d, p) not in device_property_list:
+                _to_emit.append((d, p, v))
+                super().defineConfig(group, preset, d, p, v)
+
+        self.events.newGroupPreset.emit(group, preset, _to_emit)
+
+    def defineConfigGroup(self, group: str) -> None:
+        super().defineConfigGroup(group)
+        self.events.newGroup.emit(group)
+
+    @overload
+    def defineConfig(self, group: str, preset: str) -> None:
+        ...  # pragma: no cover
+
+    @overload
+    def defineConfig(
+        self,
+        group: str,
+        preset: str,
+        device_label: str,
+        device_property: str,
+        value: str,
+    ) -> None:
+        ...  # pragma: no cover
+
+    def defineConfig(self, group: str, preset: str, *args) -> None:
+
+        if args:
+            self.defineConfigFromDevicePropertyValueList(group, preset, [args])
+        else:
+            if not self.isGroupDefined(group):
+                super().defineConfigGroup(
+                    group
+                )  # needed to refresh pymmcore 'ChannelGroup' options
+
+            if not preset:
+                idx = sum(UNNAMED_PRESET in p for p in self.getAvailableConfigs(group))
+                preset = f"{UNNAMED_PRESET}_{idx}" if idx > 0 else UNNAMED_PRESET
+
+            super().defineConfig(group, preset)
+            self.events.newGroupPreset.emit(group, preset, [])
+
+    def defineConfigFromDevicePropertyValueList(

```

Comment:
are you making this method to avoid event excess emission?

this feels more like something that an end-user should do (remember, keeping the API surface minimal is desirable)... and they can always use event blockers and/or event paused from psygnal if they want to group together event emission